sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -92,6 +92,36 @@ object PhysicalOperation extends PredicateHelper {
           .map(Alias(_, a.name)(a.exprId, a.qualifier, isGenerated = a.isGenerated)).getOrElse(a)
     }
   }
+
+  /**
+   * Drop the non-partition key expression in the disjunctions, to optimize the partition pruning.
```

Comment:
It is `(part=1 conjunction a=1) disjunction (part=2 conjunction a=4)`, right? But the expression get dropped is `a=1` which is in "conjunction" with `part=1` and `a=4` which is in "conjunction" with `part=2`. So I thought it should be conjunctions.

Or maybe we can phrase it in another way to avoid the confusion? ^_^